---
title: "Final_report"
output: html_document
date: "`r Sys.Date()`"
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = FALSE)
library(dplyr)
library(readr)
library(stringr)
library(ggplot2)
library(patchwork)
library(knitr)
library(kableExtra)
library(ComplexHeatmap)
library(BiocManager)
library(maftools)
library(GenVisR)

```

========================================================

# Exploring CPTAC Glioblastoma dataset

------------------------------------------------------------

In this project, I analyse the genomic and mutational profile of Glioblastoma cancer to identify the most common mutations and gene expression patterns.

The data set used for this project is the CPTAC Glioblastoma cohort.

The files from https://www.linkedomics.org/data_download/CPTAC-GBM/ are cleaned before using for analysis.

```{r dataset, echo=FALSE}
sampInfo <- read.csv("derived_data/sampInfo.csv")
GBM_meta <- read.csv("derived_data/GBM_metainfo.csv")
genome <- read.delim2("source_data/CPTAC_GBM_rnaseq_fpkm_uq_log2.txt")
rownames(genome)<-genome[,1]
genome<-genome[,-1]
genome<-as.data.frame(genome)

mutations<- read.delim("source_data/somatic_mutations.maf.gz")
mutations$Variant_Classification <- ifelse( mutations$Variant_Classification=="Splice_Region", "Splice_Site",mutations$Variant_Classification)
```

***

## Demographics

The data set contains information from 99 men and women aged 24 to 88, with a median age of 59. The patients are from various countries of origin, with around 30 from China, 21 from United States, 19 from Russia and 18 from Poland. 

It is interesting to note that approximately 73% of the cohort were non-smokers (62 patients). Previous studies also indicate smoking status is not a marker for glioblastoma.


![Demogrpahics plots](Figures/Demographics.png){width=50% align=center}


****

## Mutational Profile

The meta data contains information about the mutations in genes ATRX, PIK3CA, RB1, TP53, EGFR and PTEN.

It is observed that TP53 and PTEN are mutated in most samples.
Around 45 % of the cohort had only one mutation and 22% did not have any mutation. Only 1 patient had mutation in 4 genes.  

```{r mutCount}

# Subset the GBM data frame to rows 10 to 17 to choose only mutation columns
subset_data <- GBM_meta[,12:17]
subset_data <- subset_data %>% mutate_all(as.numeric)

# Calculating the sum of each column in the subset
row_sums <-rowSums(subset_data)

sum_table <- as.data.frame(table(row_sums))

# Rename the columns
colnames(sum_table) <- c("No.of.Mutations", "Freq")

# Calculating the sum of each column in the subset
col_sums <-colSums(subset_data)

col_sums<-as.data.frame(col_sums)
col_sums$genes <- rownames(col_sums)
rownames(col_sums)<-NULL
colnames(col_sums)<-c("Freq","Genes")
col_sums<-col_sums[,c("Genes","Freq")]
col_sums$Genes <- gsub("_mutation", "", col_sums$Genes)

kable(col_sums, format = "html", align = 'c', caption = "Number of Mutations in Genes") %>%
  kable_styling(full_width = FALSE, position = "center")

kable(sum_table, format = "html", align = 'c', caption = "Frequency of mutations") %>%
  kable_styling(full_width = FALSE, position = "center")

```

![Gene Counts](Figures/Gene_counts.png){align=center}

I will be focusing only on the genes ATRX, PIK3CA, RB1, TP53, EGFR and PTEN for further analysis.

```{r mutSubset}
genes <- c("ATRX", "PIK3CA", "RB1", "TP53", "EGFR", "PTEN")
idx1<-which(mutations$Hugo_Symbol %in% genes)
mutation_subset <- mutations[idx1,]

```

Gene by variant type

```{r mutTally}
mutation_subset %>% 
  group_by(Hugo_Symbol, Variant_Type) %>% 
  tally()  %>%
  arrange(desc(n))
```

![Variants per gene](Figures/Mutation_gene.png){align=center}

The package GenVisR is useful for visualizing mutation data.

![Waterfall Plot for Mutations](Figures/Mutation_waterfall.png){align=center}

*** 

## Genomic Profile


The gene expression data available is log2 transformed fpkm. 
```{r, eval=FALSE}

# Heatmap
genome <- as.data.frame(sapply(genome, as.numeric))

# Set row and column names for better readability
rownames(genomic_data) <- paste0("Gene", 1:10)
colnames(genomic_data) <- paste0("Sample", 1:10)

# Create a Heatmap object
heatmap_obj <- Heatmap(genomic_data, 
                       name = "Genomic Data", 
                       row_names_gp = gpar(fontsize = 8), 
                       column_names_gp = gpar(fontsize = 8))

# Draw the heatmap
draw(heatmap_obj, main = "Genomic Data Heatmap")

```